fhir-server/src/test/java/au/csiro/pathling/fhirpath/function/WhereFunctionTest.java

```diff
@@ -218,28 +184,33 @@ public void nullValuesAreNull() {
         .withRow("Patient/abc3", "en")
         .withRow("Patient/abc3", "en")
         .withRow("Patient/abc3", "zh")
+        .withRow("Patient/abc4", "ar")
         .build();
-    final ElementPath inputExpression = new ElementPathBuilder()
+    final ElementPath inputPath = new ElementPathBuilder()
         .fhirType(FHIRDefinedType.STRING)
         .dataset(dataset)
+        .idAndValueColumns()
         .singular(false)
         .build();
 
     // Build an expression which represents the argument to the function.
-    final Column inputCol = inputExpression.getValueColumn();
-    final Column whereResult = functions.when(inputCol.equalTo("en"), null).otherwise(true);
-    final ElementPath argumentExpression = new ElementPathBuilder()
+    final Dataset<Row> argumentDataset = inputPath.getDataset()
+        .withColumn("value",
+            functions.when(inputPath.getValueColumn().equalTo("en"), null).otherwise(true));
```

Comment:
I find it hard to understand what is being tested here. What would be the corresponding fhirpath expression?

Reply:
Well, this is really just to cover off the case where we have `null` values in the argument, particularly mixed in with other values.

Umm, as for an expression... you would probably need a nested where. It is a bit contrived, but maybe something like this:

```
Patient.communication
  .language
  .where(
    $this.communication
      .language
      .where($this.coding.code != 'en') != 'en'
  )
```